fix: tutorial (#43)
algolia/algoliasearch-netlify: README.md

  <a href="https://www.npmjs.com/package/@algolia/netlify-plugin-crawler"><img src="https://img.shields.io/npm/v/algolia/netlify-plugin-crawler" alt="Version"></a>
</p>

<h2 align="center">Algolia Netlify integration</h2>

Index your website automatically to Algolia when you deploy your project to Netlify, using our dedicated Crawler.

- [What is Algolia?](https://www.algolia.com/doc/guides/getting-started/what-is-algolia/)
- [What is Algolia's Crawler?](https://www.algolia.com/doc/tools/crawler/getting-started/overview/)

## Getting Started

On this getting started we will implement a frontend search using Netlify, Algolia, and this plugin.
To start right away **you only need a Netlify site**.

### Link your site to Algolia

**Go to <https://crawler.algolia.com/admin/netlify>**

Click "Sign-in to Algolia with Netlify".
We will automatically create a new Algolia account if you do not have one.
<img src="/docs/screenshots/screely-1601375945482.png?raw=true" alt="Sign in to Algolia with Netlify">

Authorize Algolia to access your Neltify account.
The plugin will only update your plugin settings and add two environement variables.

> NB: Algolia **only stores your encrypted Netlify access token**, and no other information.

Author: I was thinking about the other private env variable that we might have access too (but do not store), but I'll remove until we find it necessary

> NB: Algolia **only stores your encrypted Netlify access token**, and no other information.

Reviewer: Ha yes, I understand it more. The cool part about it being open source is that it's easily inspectable that we do not transmit more than needed.

> NB: Algolia **only stores your encrypted Netlify access token**, and no other information.

Author: well the plugin yes, but not our backend 😬

<img src="/docs/screenshots/screely-1601375955283.png?raw=true" alt="Authorize Algolia's oauth">

Search your site and click "install".
On this step we will automatically update your Netlify site to add a couple of environment variables required to use the plugin.
We will also create one Algolia application with a dedicated [free plan](https://www.algolia.com/pricing/).

<img src="/docs/screenshots/screely-1601375970172.png?raw=true" alt="Search and install your site.">
<img src="/docs/screenshots/screely-1601375981021.png?raw=true" alt="Search and install your site.">

The plugin is now correctly installed and should already index your site at the next deploy.
<img src="/docs/screenshots/screely-1601375010606.png?raw=true" alt="View your site information.">

### Build your index

You can deploy your site to Netlify and wait for your index to be populated by the Crawler.
If you want, in the meantime, you can also use our own [Frontend Bundle](#install-the-frontend-bundle) to power your search out of the box

> Once we receive a build hooks, the Crawler will process your website asynchronously. Indeed the Crawler requires your site to be online to be able to crawl it. In other words, there will be a small delay between deploying and having your Algolia index populated.

Reviewer: ```suggestion
Once we receive a build hook, our Crawler will process your website asynchronously, and this operation takes some time.
In other words, there will be a delay between the first deploy on a branch and the Algolia index being ready for usage.
```

> In Netlify deploy logs you will find the information about your current Crawler.

Reviewer: ```suggestion
You will find the information about your current Crawler in the Netlify deploy logs.
```

> We will create **one Crawler and one Algolia index per branch name**, so you can have a production index and develop index for example.

Reviewer: ```suggestion
We will create **one Crawler targeting one Algolia index per git branch**, so you can have a production index on `master` and development index on `develop` for example.
```

<img src="/docs/screenshots/screely-1601381176485.png?raw=true" alt="Netlify deploy logs.">

> You can click on the URL to go the current Crawlergo to our UI to follow the progress of the crawl.

Reviewer: ```suggestion
You can click on the URL to open the current Crawler dashboard to follow the progress of the crawl.
```

<img src="/docs/screenshots/screely-1601381447204.png?raw=true" alt="Your Crawler running.">

> After the crawl is done you can go take a look at your Algolia index, to see the records we extracted automatically.

Reviewer: ```suggestion
After the crawl is done you can go take a look at your Algolia index to see the records we extracted automatically.
```

> We apply a default relevance by default, but you can fine tune it as you want in the index settings.

Reviewer: ```suggestion
We apply a default relevance configuration by default, but you can fine tune it as you want in the index settings.
```

Find out [what is inside your records](/docs/schema.md).
<img src="/docs/screenshots/screely-1601382688193.png?raw=true" alt="Your Algolia Index.">

### Install the frontend bundle

> Note: we recommend using our bundle because it is compatible with the records we extract from your website.
> You are free to build the UI you want, with [InstantSearch.js](https://www.algolia.com/doc/guides/building-search-ui/what-is-instantsearch/js/) or any other frontend technology.

In the Crawler UI you will find an html snippet that you can copy into your code.
Replace the variables with the one given in the UI:

- `YOUR_ALGOLIA_APP_ID`: the unique identifier of your Algolia Application
- `YOUR_ALGOLIA_API_KEY`: the Algolia Search-Only api key (you can found it here <https://www.algolia.com/api-keys>)
- `YOUR_BRANCH` with your branch name (any branch name where the plugin is allowed to run)

```html

> <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/@algolia/algoliasearch-netlify-frontend/dist/algoliasearchNetlify.css" />
> <script type="text/javascript" href="https://cdn.jsdelivr.net/npm/@algolia/algoliasearch-netlify-frontend/dist/algoliasearchNetlify.js"></script>

Reviewer: ```suggestion
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/@algolia/algoliasearch-netlify-frontend@0/dist/algoliasearchNetlify.css" />
<script type="text/javascript" href="https://cdn.jsdelivr.net/npm/@algolia/algoliasearch-netlify-frontend@0/dist/algoliasearchNetlify.js"></script>
```

<script type="text/javascript">
  algoliasearchNetlify({
    appId: '<YOUR_ALGOLIA_APP_ID>',
    apiKey: '<YOUR_ALGOLIA_API_KEY>',
    indexName: 'netlify_<YOUR_NETLIFY_SITE_ID>_<YOUR_BRANCH>',
  });
</script>
```

> This will automatically plug your newly created Algolia index in your website.

Reviewer: ```suggestion
This will automatically add an instant search autocomplete to your website using your newly created Algolia index.
```

Please refer to the [full documentation](https://github.com/algolia/algoliasearch-netlify/tree/master/frontend) to configure this frontend plugin.

> Screenshot

### Going further

- Understand the extracted [record schema](/docs/schema.md)
- Learn more about [Algolia](https://www.algolia.com/doc/)
- Implement a dedicated UI with [InstantSearch.js](https://www.algolia.com/doc/guides/building-search-ui/what-is-instantsearch/js/)
- Fine tune your search relevancy [Relevance Guide](https://www.algolia.com/doc/guides/managing-results/relevance-overview/)

## Uninstall

You can uninstall the plugin by either:

- Go to your Crawler UI and click "uninstall"
  **This is the recommended way**.
  We will also clean the environment variables and delete associated data from Algolia.
- Go to your Netlify plugin and click "uninstall"
  This will not clean your data.

> ## Limitation / Known Issues / FAQ

Reviewer: ```suggestion
## Limitations / Known Issues / FAQ
```

> ### My website requires Javascript to run

Reviewer: ```suggestion
### My website requires JavaScript to run
```

The crawler does support this but the option is not yet exposed in the plugin.
Please contact us to enable it.

### I want to extract more information from my webpages

While the Crawler is extremely customisable, the plugin is currently restricted to a default extraction strategy.

> If you believe we have missed something or you would want to have more control, please share your feedback in our [Discourse](https://discourse.algolia.com/c/netlify/28)

Reviewer: ```suggestion
If you believe we have missed something or you would want to have more control, please share your feedback in our [Discourse forum](https://discourse.algolia.com/c/netlify/28).
```

> ### Can I build my own UI

Reviewer: ```suggestion
### Can I build my own UI?
```

> Yes !

Reviewer: ```suggestion
Yes!
```

> ### Can I receive a notification when my crawl is done

Reviewer: ```suggestion
### Can I receive a notification when my crawl is done?
```

Yes, you can configure notification in the Crawler UI here: <https://crawler.algolia.com/admin/user/settings/>

> ## Troubleshoot

Reviewer: ```suggestion
## Troubleshooting
```

> - Need help? We have you covered in our [Discourse](https://discourse.algolia.com/c/netlify/28)

Reviewer: ```suggestion
- Need help? We have you covered in our [Discourse forum](https://discourse.algolia.com/c/netlify/28)
```

> - Found a bug in the plugin? Please post an issue in this repository and read our [Contributing guide](/CONTRIBUTING.md)

Reviewer: ```suggestion
- Found a bug in the plugin? Please read our [Contributing guide](/CONTRIBUTING.md) and post an issue in this repository
```
